pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixSegmentOnlineOfflineStateModelGenerator.java

```diff
@@ -46,6 +40,10 @@ private PinotHelixSegmentOnlineOfflineStateModelGenerator() {
   public static final String DROPPED_STATE = "DROPPED";
   public static final String CONSUMING_STATE = "CONSUMING";
 
+  public static final int DEFAULT_TRANSITION_PRIORITY = 1000;
+  // Given a lower priority to let the consuming segment start last during server startup.
+  public static final int OFFLINE_TO_CONSUMING_TRANSITION_PRIORITY = 1100;
```

Comment:
Why such high values? I guess you can make the default as 1 and offline to consuming as 2?

Reply:
just give enough buffer between each state.

Reply:
what do you mean by buffer though?

Reply:
In case we want to insert other state with priority in between.